python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,457 @@
+    Point,
+    MultiPoint,
+    LineString,
+    MultiLineString,
+    Polygon,
+    MultiPolygon,
+)
+
+import cudf
+from cudf.core.column import ColumnBase
+
+from cuspatial.io.geoseries_reader import GeoSeriesReader
+
+
+class GeoSeries(ColumnBase):
+    def __init__(self, data, interleaved=True):
+        """
+        A GPU GeoSeries object.
+
+        Parameters
+        ----------
+        data : A GeoPandas GeoSeries object, or a file path
+
+        interleaved : Boolean
+            Store x,y coordinates in an interleaved [x,y] pattern according
+            to GeoArrow form or store x and y coordinates in separate
+            parallel buffers. Defaults to True.
+        Discussion
+        ----------
+        The GeoArrow format specifies a tabular data format for geometry
+        information. Supported types include Point, MultiPoint, LineString,
+        MultiLineString, Polygon, and MultiPolygon.  In order to store
+        these coordinate types in a strictly tabular fashion, columns are
+        created for Points, MultiPoints, LineStrings, and Polygons.
+        MultiLines and MultiPolygons are stored in the same data structure
+        as LineStrings and Polygons.
+
+        The Points column is a simple numeric column. x and y coordinates
+        can be stored either interleaved or in separate columns. If a z
+        coordinate is present, it will be stored in a separate column.
+
+        The MultiPoints column is similar to the Points column with the
+        addition of an offsets column. The offsets column stores the comparable
+        sizes and coordinates of each MultiPoint in the cuGeoSeries.
+
+        LineStrings contain the coordinates column, an offsets column, and a
+        multioffsets column. The multioffsets column stores the indices of the
+        offsets that indicate the beginning and end of each MultiLineString
+        segment.
+
+        Polygons contain the coordinates column, a rings column specifying
+        the beginning and end of every polygon, a polygons column specifying
+        the beginning, or exterior, ring of each polygon and the end ring.
+        All rings after the first ring are interior rings.  Finally a
+        multipolys column stores the offsets of the polygons that should be
+        grouped into MultiPolygons.
+
+        As a result, a GpuGeoSeries object contains these arrow-supported
+        columns:
+        points_xy
+        mpoints_xy
+        mpoints_offsets
+        lines_xy
+        lines_offsets
+        lines_ml_offsets
+        polygons_xy
+        polygons_rings
+        polygons_polys
+        polygons_mpolys
+
+        Notes
+        -----
+        Legacy cuspatial algorithms depend on separated x and y columns.
+        Creating a cuGeoSeries from a GeoPandas source with `interleaved=True`
+        will create a legacy cuspatial Series, which violates the GeoArrow
+        format but will work with existing cuspatial algorithms.
```

Comment:
This is false. I'm going to drop the constructor argument and instead provide a `cuGeoSeries.de_interleave()` method to separate the x/y columns for use in our existing algs.